### DISCUSSION TITLE
Auto-generated API route for `useDeno`

### DISCUSSION THREAD

**Discussion by rliang** [alephjs/aleph.js#225] [Ideas]
In Next.js, `getServerSideProps` is called server-side on each refresh, and it is able to access Node.js APIs. In Aleph, `useDeno(..., true)` is called on each refresh, but not server-side. I suppose I could get around that by adding an API route and have `useDeno` fetch it.

However, I think it would be a big productivity bonus if the hook could always access the backend. Perhaps by having Aleph auto-generate an API route, even.

If possible, would you be willing to consider adding this functionality?

**Comment by aadamsx** [comment]
Your callback will run in the browser (not use cached data) by passing `true` as the second parameter in `useDeno`: https://alephjs.org/docs/advanced-features/use-deno-hook

```javascript
const post = useDeno(async () => {
  return await (await fetch(`https://.../post/${params.id}`)).json();
}, true);
```

Why would you want to make a round trip to the server to get data, instead of making a localized async call to an API?  What's the advantage or use case?

**Comment by shadowtime2000** [comment]
@rliang I think @ije added ISR (incremental static regeneration) which may be what you are looking for.

**Comment by ije** [comment]
in 0.3 branch, the second argument is `revalidate` that means the **lifetime** for the SSR data in seconds. the page will be re-render on the server client when it is expired. and the first argument `callback` will be removed on the client.
but yes! we need something like `getServerSideProps`, i'm still think about it 🤯

**Comment by rliang** [comment]
Ah yes, I didn't realize that the second argument changed. Wouldn't setting revalidate to 0 make it work like `getServerSideProps`?
